Added Oven and Stove Blocks

TheeMonster395 opened the pull request.

TheeMonster395 committed:
Added Oven and Stove Blocks

-Added oven block
-Added stove block

TheeMonster395 committed:
Merge branch 'BedrockCommands:main' into main

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/BedrockCommands/Cook-Off?pullRequest=14) <br/>All committers have signed the CLA.

jeanmajid reviewed (changes requested):
this should be bcc.cook:stove

jeanmajid commented on behavior_pack/blocks/bcc.cook/stove.json:
```diff
@@ -0,0 +1,25 @@
+					"texture": "stove"
+				}
+			},
+            "bcc.cook:stove_behavior": {}
```
this should be bcc.cook:stove

jeanmajid commented on behavior_pack/src/components/blocks/stove.ts:
```diff
@@ -0,0 +1,62 @@
+            return
+        }
+
+        /**@type {import("@minecraft/server").ItemStack} */
```
Use ts types, not jsdoc

jeanmajid reviewed (changes requested):
Make sure you use semicolons and you get rid of unused imports

TheeMonster395 committed:
Resolved issues

jeanmajid commented on behavior_pack/src/components/blocks/stove.ts:
```diff
@@ -0,0 +1,60 @@
+    "bcc.cook:pot": { sound: "sound of placing pot" }
+}
+
+function addVectors(a, b) {
```
Use the the Vector class instead of making your own functions

jeanmajid commented on behavior_pack/src/components/blocks/stove.ts:
```diff
@@ -0,0 +1,60 @@
+import Utils from "../../Utils"
+import SoundManager from "../../SoundManager"
+import { BlockId } from "../../constants/blockId"
+import { Dimension, ItemStack } from "@minecraft/server"
```
remove unused imports

jeanmajid commented on behavior_pack/src/components/blocks/stove.ts:
```diff
@@ -0,0 +1,60 @@
+})
+
+
+const canPlaceOnStove = {
```
move the definition of this object above the actual stove registry

TheeMonster395 replied:
Do you mean below the imports and above the registry?

TheeMonster395 committed:
Added cutting board and sink

-Added Cutting Board block
-Added Sink block
-Added rotation when placed to Oven and Stove

TheeMonster395 committed:
Merge branch 'main' into main

jeanmajid reviewed (changes requested):
sorry for the late review, I forgot to submit it

TheeMonster395 committed:
Made changes in stove.ts

TheeMonster395 committed:
Merge branch 'main' of https://github.com/TheeMonster395/Cook-Off

TheeMonster395 committed:
tranpiled shanges
